docs/experimental/examples/oauth2/globus_app.rst

```diff
@@ -1,5 +1,202 @@
 
-Using a Globus App
-==================
+.. _using_globus_app:
 
-TODO
+.. py:currentmodule:: globus_sdk.experimental.globus_app
+
+Using a GlobusApp
+=================
+
+Programmatic communication with Globus services relies on the authorization of requests.
+Management and resolution of this authorization can become an arduous task, especially
+when a script needs to interact with different services each carrying an individual set
+of complex auth requirements. To assist with this task, this library provides a utility
+construct called a GlobusApp.
+
+A :py:class:`~GlobusApp` is a distinct object which will manage auth requirements
+(i.e. **scopes**) identified by their associated service (i.e. **resource server**).
+In addition to storing these requirements, a GlobusApp provides a mechanism to resolve
+unmet ones through browser and api-based auth flows, supplying the resulting tokens to
+bound clients as requested.
+
+There are two flavors of GlobusApp:
+
+*   :py:class:`~UserApp`, a GlobusApp for interactions between an end user and Globus
+    services. Facilitates interactions *on behalf of a user*.
+
+*   :py:class:`~ClientApp`, a GlobusApp for interactions between a client
+    (i.e. service account) and Globus services. Facilitates interactions
+    *as the script itself*.
+
+Setup
+-----
+
+A GlobusApp is heavily configurable object. For common scripting usage however,
+instantiation only requires two parameters:
+
+#.  **App Name** - A human readable slug to identify your app in http requests and token
```

Comment:
The regex you described is not required but it is recommended.

We'll be using this to generate a file name so I'd trying to push users in the direction of "my-cool-app" instead of "My App".

Reply:
If the user selects an in-memory token store, is that still a consideration?

My assumption here is that it's more important how this appears in the Web App when consenting to things, so I'd prefer to slugify the app name when creating a file-backed token store, but not use the word "slug" here.

Reply:
If a user provides their own instantiated token storage, in-memory or not, this value is unused.

At this point we're debating behavior though not documentation.
Can we remove this from the documentation PR & discuss out-of-band whether to make the change you're suggesting?